control/shift_decider/src/shift_decider.cpp

```diff
@@ -83,7 +83,7 @@ void ShiftDecider::updateCurrentShiftCmd()
     } else if (control_cmd_->longitudinal.speed < -vel_threshold) {
       shift_cmd_.command = GearCommand::REVERSE;
     } else {
-      shift_cmd_.command = current_gear_ptr_->report;
+      shift_cmd_.command = prev_shift_command;
```

Comment:
Reading the PR background, the problem is occurred when the ego engages, which means shifting from parking to driving.
In that situation the change should be appried not at L86 but L95?

Reply:
```shift_cmd_.command = current_gear_ptr_->report``` at L95 is called when the vehicles does not engages.
I think L95 is OK to be as it is

Reply:
Got it thank you.